### ISSUE TITLE
`dev::clickhouse::tests::test_discover_local_listening_port_slow_write` is flaky

### ISSUE THREAD

**Issue by smklein** [oxidecomputer/omicron#527]
On my Linux machine, with https://github.com/oxidecomputer/omicron/pull/526 patched so I can run isolated tests, I am running the following:

```bash
set -eu

for i in $(seq 1 1000)
do
  echo "Iteration $i"
  cargo test -p omicron-test-utils --lib
done
```

And within a few iterations (usually within less than 10) I see the following:

```
Starting reader task                                                                                                    
Writing to log file                                                                                                     
Advancing time by 20ms                                                                                                  
Advancing time by 20ms                                      
Advancing time by 20ms                                      
Writing to log file                                                                                                     
Advancing time by 20ms                                                                                                  
Advancing time by 20ms                                                                                                  
Advancing time by 20ms                                                                                                  
Writing to log file                                                                                                     
Advancing time by 20ms                                                                                                  
Advancing time by 20ms                                                                                                  
Advancing time by 20ms                                                                                                  
Writer finished                                                                                                         
Advancing time by 20ms                                                                                                  
Advancing time by 20ms                                                                                                  
Advancing time by 20ms                                                                                                  
Advancing time by 20ms                                                                                                  
Advancing time by 20ms                                      
Advancing time by 20ms                                                                                                  
Advancing time by 20ms                                                                                                  
Advancing time by 20ms                   
...
Advancing time by 20ms
Advancing time by 20ms
Reader finished
thread 'dev::clickhouse::tests::test_discover_local_listening_port_slow_write' panicked at 'called `Result::unwrap()` on an `Err` value: Timeout', test-utils/src/dev/clickhouse.rs:301:70
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by smklein** [comment]
FWIW, this is using `ClickHouse server version 21.7.1.1.`

**Comment by bnaecker** [comment]
Thanks for filing this! I think this is basically just due to scheduling jitter. The test simulates ClickHouse writing to a log file while we're looking for the port number in it. The reader task is one branch of a `tokio::select!` loop, and it just seems to end up always being cancelled when these tests fail.

I think my understanding of tokio's test timer facilities also got in the way. The test tries to control time closely, but that's really not necessary. I've removed that bit, and things are still passing 20 iterations into your test loop.

There's a fix up in #531

**Comment by bnaecker** [comment]
Closed by #531
